cli/pcluster/config/validators.py

```diff
@@ -839,3 +839,41 @@ def intel_hpc_validator(param_key, param_value, pcluster_config):
         )
 
     return errors, warnings
+
+
+def custom_chef_cookbook_validator(param_key, param_value, pcluster_config):
+    """
+    Validate custom_chef_cookbook parameter.
+
+    Assumes that the S3 bucket where the cookbook is stored should be listed in
+    s3_read_resource explicitly, or s3_read_resource should contain a wildcard.
+    """
+    errors = []
+    warnings = []
+
+    if urlparse(param_value).scheme in ["https", "http"]:
+        errors, warnings = url_validator(param_key, param_value, pcluster_config)
+
+    elif urlparse(param_value).scheme == "s3":
+
+        # Check that bucket is in s3_read_resource.
+        cluster_section = pcluster_config.get_section("cluster")
+        s3_read_resource = cluster_section.get_param_value("s3_read_resource")
+
+        if s3_read_resource == "*":
+            pass
+        else:
+            match = re.match(r"s3://(.*?)/(.*)", param_value)
+            if not match or len(match.groups()) < 2:
+                errors.append("S3 url is invalid.")
+            bucket = match.group(1)
+
+            # Parse bucket name from S3 URI.
+            # Match anything after arn prefix, until it hits / or *.
+            match = re.search(r"(?<=arn:aws:s3:::)([^*/]*)", s3_read_resource)
+            if bucket != match.group(0):
+                errors.append("Bucket {} is not in your s3_read_resource parameter.".format(bucket))
```

Comment:
Instead of creating an error, I'd use a warning, like we're doing in the url validator because for example the user could upload the cookbook in a bucket and make it public or use a policy to access to it.

Reply:
Done.